### PR TITLE
Issue-26: Update test_health.gd to use the cards, fix a bug with Damage All card by adding EffectDamageAll.gd

### PR DESCRIPTION
**Pull request by Tysterman74**
# Description

Updated test_health.gd and added a few unit tests that test around using the actual cards in the Cards/Resource folder and test their effects. Fixed a bug with Damage All card.

## Related issue(s)

#26 

## List of changes

- Updated test_health.gd, added new unit tests that tests each card currently in Cards/Resource folder
- Added EffectDamageAll.gd to fix a bug of Damage All Card not working
- Added the new Effect to the Card_DamageAll.tres
- Updated the test scene to include in the Card_DamageAll in the default deck

## Tests

- Updated test_health.gd, added new unit tests that tests each card currently in Cards/Resource folder

## Additional notes

There may need to be an update to the poison tests since currently there hasn't been a decision on what the poison should be doing in terms of damage. Left a comment there to elaborate.

### CODE REVIEW

**Comment by Turtyo** [comment]
I think we will have conflicts with #39 once merged because it changes DamageAll too iirc

File: Tests/test_health.gd [hunk 1]
```diff
@@ -149,3 +153,64 @@ func test_poison_status():
...
+	assert_is(status, Debuff_Poison)
+	assert_eq(status.status_turn_duration, 3)
+	
+	_enemy.get_status_component().apply_turn_start_status()
+	# May need to update once we have a better direction of what to do for poison, currently
+	# it deals only 1 damage per turn
+	assert_eq(_enemy_health_component.current_health, 99.0)
+
+func test_card_damage_and_poison():
+	var card_damage_and_poison: CardBase = load("res://Cards/Resource/Card_damage_and_poison.tres")
```
Comment (Turtyo) [inline Tests/test_health.gd:195]: maybe adding the description of the card in the test would be good (so we don't need to go check the card)

**Review by Turtyo — CHANGES_REQUESTED** [review]
Not much to say, it looks clean :+1:

File: Tests/test_health.gd [hunk 2]
```diff
@@ -191,6 +198,7 @@ func test_card_poison():
 	# it deals only 1 damage per turn
 	assert_eq(_enemy_health_component.current_health, 99.0)
 
+# Test Card that deals 1 damage and applies poison to enemy and does poison damage on enemy turn
```
Comment (Turtyo) [inline Tests/test_health.gd:201]: missing number of poison stack

**Review by Turtyo — APPROVED** [review]

**Review by Turtyo — APPROVED** [review]

**Review by cheesefrycook — APPROVED** [review]

**Pull request merged**